lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -217,13 +218,19 @@ IncrementalJIT::getSymbolAddressWithoutMangling(llvm::StringRef Name,
   if (auto Sym = getInjectedSymbols(Name))
     return Sym;
 
+  const std::string SymName = Name;
```

Comment:
Everything past that line takes a std::string so it'd still be good to cache the conversion.

CStr variable?
I agree it's a nicer object and thought to use a StringRef, but currently all code using DLOpen/DLClose is a std::string.
It also seems more succiint as the optional err parameter will need to be a std::string and then you've got to eat the include anyway.